test/internals/request.test.js

```diff
@@ -266,3 +273,470 @@ test('Request with trust proxy and undefined socket', t => {
   const request = new TpRequest('id', 'params', req, 'query', 'log')
   t.same(request.protocol, undefined)
 })
+
+test('Request#validate', subtest => {
+  const defaultSchema = {
+    type: 'object',
+    required: ['hello'],
+    properties: {
+      hello: { type: 'string' },
+      world: { type: 'string' }
+    }
+  }
+  const requestSchema = {
+    params: {
+      id: {
+        type: 'integer',
+        minimum: 1
+      }
+    },
+    querystring: {
+      foo: {
+        type: 'string',
+        enum: ['bar']
+      }
+    },
+    body: defaultSchema,
+    headers: {
+      'x-foo': {
+        type: 'string'
+      }
+    }
+  }
+
+  subtest.plan(6)
+
+  subtest.test('Should return a function when empty input - Route without schema', async t => {
+    const fastify = Fastify()
+
+    t.plan(3)
+
+    fastify.get('/', (req, reply) => {
+      const validate = req.validate(defaultSchema)
```

Comment:
Isn't this `req.compile`?

Reply:
The goal is more or less to give the flexibility to either compile the function or validate on-the-fly. With this, the implementation has the possibility to compile the function and share it across the request lifetime or just execute a one-time validation by providing the schema.

Either way, the compiled function is cached through a `WeakMap` coupled to the provided schema. Still not sure if this is the final shape, so of course open to feedback to adjust it 🙂

Reply:
It seems the method returns a function that does the validation, and does not do any validation itself. Thus, it seems misnamed.

Reply:
sgtm, let me adjust it 👍